src/python/module/smartredis/client.py

```diff
@@ -188,11 +195,13 @@ def copy_dataset(self, key, dest_key):
         :type dest_key: str
         :raises RedisReplyError: if copy operation fails
         """
-        try:
-            super().copy_dataset(key, dest_key)
-        except RuntimeError as e:
-            raise RedisReplyError(str(e), "copy_dataset") from None
+        if not isinstance(key, str):
```

Comment:
So I think we can reduce these lines down to a simpler function like

```python
In [2]: def type_check(arg, name="key", _type=str):
   ...:     if not isinstance(arg, _type):
   ...:         raise TypeError(f"Argument {name} was of type {type(arg)}, not {_type}")
```

Some example uses
```python
In [3]: type_check("hello") # no error raised

In [4]: type_check(1) 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [4], in <module>
----> 1 type_check(1)

Input In [2], in type_check(arg, name, _type)
      1 def type_check(arg, name="key", _type=str):
      2     if not isinstance(arg, _type):
----> 3         raise TypeError(f"Argument {name} was of type {type(arg)}, not {_type}")

TypeError: Argument key was of type <class 'int'>, not <class 'str'>
```

example with different name
```
In [6]: type_check(1, name="data")
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [6], in <module>
----> 1 type_check(1, name="data")

Input In [2], in type_check(arg, name, _type)
      1 def type_check(arg, name="key", _type=str):
      2     if not isinstance(arg, _type):
----> 3         raise TypeError(f"Argument {name} was of type {type(arg)}, not {_type}")

TypeError: Argument data was of type <class 'int'>, not <class 'str'>

In [7]:
```

So instead of each of these methods looking like

```python
    @exception_handler
    def copy_tensor(self, key, dest_key):
        """Copy a tensor at one key to another key

        :type dest_key: str
        :raises RedisReplyError: if copy operation fails
        """
        if not isinstance(key, str):
            raise TypeError(f"Argument provided for key, {type(key)}, is not of type str")
        if not isinstance(dest_key, str):
            raise TypeError(f"Argument provided for dest_key, {type(dest_key)}, is not of type str")
        super().copy_tensor(key, dest_key)
```

they would instead read as

```python

    @exception_handler
    def copy_tensor(self, key, dest_key):
        """Copy a tensor at one key to another key

        :type dest_key: str
        :raises RedisReplyError: if copy operation fails
        """
        type_check(key)
        type_check(dest_key, name="dest_key")
        super().copy_tensor(key, dest_key)
```

We can place this ``type_check`` function in utils and make sure to add a comment about not wanting to expose the pybind layer errors.

Reply:
This should also make it so that you don't have to have a test for each type error to obtain full coverage

Reply:
Awesome, I wanted to do this and couldn't get the function working correctly